src/main/java/org/minions/devfund/abel/Squarelotron.java

```diff
@@ -0,0 +1,184 @@
+        this.squarelotronMatrix = deepCopy(squarelotron);
+        this.size = this.squarelotronMatrix.length;
+    }
+
+    /**
+     * Get the Squarelotron.
+     *
+     * @return The Squarelotron.
+     */
+    public int[][] getSquarelotronMatrix() {
+        return deepCopy(squarelotronMatrix);
+    }
+
+    /**
+     * This method performs the Upside-Down Flip of the squarelotronMatrix, as described above,
+     * and returns the new squarelotronMatrix.
+     *
+     * @param ring The ring to be modified.
+     * @return A new Squarelotron object.
+     */
+    public Squarelotron upsideDownFlip(int ring) {
+        Function<Integer, Integer, Integer> upsideDown = (i, j) -> squarelotronMatrix[size - i - 1][j];
+        return new Squarelotron(flip(ring, upsideDown));
+    }
+
+    /**
+     * This method performs the Main Diagonal Flip of the squarelotronMatrix, as described above,
+     * and returns the new squarelotronMatrix.
+     *
+     * @param ring The ring to be modified.
+     * @return A new Squarelotron object.
+     */
+    public Squarelotron mainDiagonalFlip(int ring) {
+        Function<Integer, Integer, Integer> upsideDown = (i, j) -> squarelotronMatrix[j][i];
+        return new Squarelotron(flip(ring, upsideDown));
+    }
+
+    /**
+     * Method to flip based on a Function.
+     *
+     * @param ring The ring to be modified.
+     * @param fn   The lambda function to execute.
+     * @return 2d array flipped.
+     */
+    private int[][] flip(int ring, final Function<Integer, Integer, Integer> fn) {
+        int[][] copy = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                if (isInRing(ring, i, j)) {
+                    copy[i][j] = fn.apply(i, j);
+                } else {
+                    copy[i][j] = squarelotronMatrix[i][j];
+                }
+            }
+        }
+        return copy;
+    }
+
+    /**
+     * Custom Functional interface.
+     *
+     * @param <T>  The first parameter.
+     * @param <T2> The second parameter.
+     * @param <R>  The result response.
+     */
+    @FunctionalInterface
+    interface Function<T, T2, R> {
+        /**
+         * Method to execute the function.
+         *
+         * @param one First parameter.
+         * @param two Second parameters.
+         * @return The response type.
+         */
+        R apply(T one, T2 two);
+    }
+
+    /**
+     * The argument numberOfTurns indicates the number of times the entire squarelotronMatrix
+     * should be rotated 90° clockwise.
+     * Any integer, including zero and negative integers, is allowable as the argument.
+     * A value of -1 indicates a 90° counterclockwise rotation.
+     *
+     * @param numberOfTurns The times to rotate.
+     */
+    public void rotateRight(int numberOfTurns) {
+        int[][] copy = new int[size][size];
+        for (int x = 0; x < Math.abs(numberOfTurns); x++) {
```

Comment:
if the numberOfTurns is 4 or 8, it should return the same matrix.
if the numberOfTurns is 5 or 9, it should rotate only one time because of the first scenario.